docs/Ramm.md

```diff
@@ -0,0 +1,186 @@
+Several constants manage the liquidity behavior and price adjustments:
+
+```solidity
+uint public constant LIQ_SPEED_PERIOD = 1 days;
+uint public constant TARGET_LIQUIDITY = 5_000 ether;
+uint public constant FAST_LIQUIDITY_SPEED = 1_500 ether;
+uint public constant NORMAL_RATCHET_SPEED = 400;
+uint public constant FAST_RATCHET_SPEED = 5_000;
+```
+
+## Mutative Functions
+
+### `swap`
+Allows users to swap NXM tokens for ETH or vice versa.
+
+```solidity
+function swap(
+  uint nxmIn,
+  uint minAmountOut,
+  uint deadline
+) external payable nonReentrant returns (uint);
+```
+
+| Parameter        | Description                                                                                  |
+|------------------|----------------------------------------------------------------------------------------------|
+| `nxmIn`          | Amount of NXM to swap for ETH (0 if swapping ETH for NXM, and send ETH with the transaction) |
+| `minAmountOut`   | Minimum amount of ETH or NXM expected from the swap                                          |
+| `deadline`       | The deadline for the swap to be executed                                                     | 
+
+### `removeBudget`
+Resets the ETH budget used for liquidity injection to zero. This function can only be called by governance.
+
+```solidity
+function removeBudget() external onlyGovernance;
+```
+
+### `setEmergencySwapPause`
+Allows an emergency administrator to pause or resume swap functionality during emergencies.
+
+```solidity
+function setEmergencySwapPause(bool _swapPaused) external onlyEmergencyAdmin;
+```
+
+| Parameter      | Description                                |
+|----------------|--------------------------------------------|
+| `_swapPaused`  | True to pause swaps, false to resume them  |
+
+### `setCircuitBreakerLimits`
+Sets the limits for the circuit breakers, controlling the maximum ETH and NXM that can be released.
+
+```solidity
+function setCircuitBreakerLimits(
+  uint _ethLimit,
+  uint _nxmLimit
+) external onlyEmergencyAdmin;
+```
+
+| Parameter    | Description                                  |
+|--------------|----------------------------------------------|
+| `_ethLimit`  | Maximum ETH that can be released             |
+| `_nxmLimit`  | Maximum NXM that can be released             |
+
+## View Functions
+
+### `getReserves`
+Returns the current reserves and budget of the contract.
+
+```solidity
+function getReserves() external view returns (
+  uint ethReserve,
+  uint nxmA,
+  uint nxmB,
+  uint budget
+);
+```
+
+| Return         | Description                         |
+|----------------|-------------------------------------|
+| `ethReserve`   | ETH reserve in the contract         |
+| `nxmA`         | NXM buy price                       |
+| `nxmB`         | NXM sell price                      |
+| `budget`       | Available ETH budget for liquidity  |
+
+### `getSpotPrices`
+Returns the current spot prices for NXM buy and sell operations.
```

Comment:
Worth mentioning that when swapping the prices are on a uniswap v2 -like curve and the spot prices are just where the price curve starts so someone wouldn't expect to multiply this by the amount and expect that result.